sdk/metrics/src/debugEnabledTest/java/io/opentelemetry/testing/TestSourceInfo.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.testing;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import io.opentelemetry.sdk.metrics.common.InstrumentType;
+import io.opentelemetry.sdk.metrics.common.InstrumentValueType;
+import io.opentelemetry.sdk.metrics.internal.debug.SourceInfo;
+import io.opentelemetry.sdk.metrics.internal.descriptor.InstrumentDescriptor;
+import io.opentelemetry.sdk.metrics.internal.descriptor.MetricDescriptor;
+import io.opentelemetry.sdk.metrics.internal.state.DebugUtils;
+import io.opentelemetry.sdk.metrics.view.View;
+import org.junit.jupiter.api.Test;
+
+// Note: This class MUST be outside the io.opentelemetry.metrics package to work correctly.
```

Comment:
Why?

Reply:
Because we skip any thing in the stack in `io.opentelemetry.metrics` package

Reply:
We're truncating the "io.opentelemetry.sdk.metrics" package stack traces when we dump Instrument registration to avoid useless context.  I.e. we try to grab the stack trace where an instrument was registered, but deep from the bowels of the SDK.

Reply:
Right 👌